src/main/java/org/mskcc/oncokb/curation/config/ElasticsearchConfiguration.java

```diff
@@ -0,0 +1,117 @@
+package org.mskcc.oncokb.curation.config;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import org.mskcc.oncokb.curation.converters.EntityToMapConverter;
+import org.mskcc.oncokb.curation.converters.MapToEntityConverter;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.data.convert.WritingConverter;
+import org.springframework.data.elasticsearch.config.ElasticsearchConfigurationSupport;
+import org.springframework.data.elasticsearch.core.convert.ElasticsearchCustomConversions;
+
+@Configuration
+public class ElasticsearchConfiguration extends ElasticsearchConfigurationSupport {
+
+    private final EntityToMapConverter entityToMapConverter;
+    private final MapToEntityConverter mapToEntityConverter;
+
+    public ElasticsearchConfiguration(EntityToMapConverter entityToMapConverter, MapToEntityConverter mapToEntityConverter) {
+        this.entityToMapConverter = entityToMapConverter;
+        this.mapToEntityConverter = mapToEntityConverter;
+    }
+
+    @Bean
+    @Override
+    public ElasticsearchCustomConversions elasticsearchCustomConversions() {
+        return new ElasticsearchCustomConversions(
+            Arrays.asList(
+                new ZonedDateTimeWritingConverter(),
+                new ZonedDateTimeReadingConverter(),
+                new InstantWritingConverter(),
+                new InstantReadingConverter(),
+                new LocalDateWritingConverter(),
+                new LocalDateReadingConverter(),
+                entityToMapConverter,
+                mapToEntityConverter
```

Comment:
The date converters are generated by jhipster because elasticsearch doesn't have date types. Instead they should be converted to strings.

I added `entityToMapConverter` and `mapToEntityConverter` to resolve the stackoverflow error caused by circular associated entities. [Spring data elasticsearch since 4.0 removed jackson object mapper](https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#elasticsearch-migration-guide-3.2-4.0.jackson-removal), so these converters is a workaround.